lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+    return shape;
+  };
+
+  /**
+   * Update this collider based on the properties of a parent Sprite.
+   * @method updateFromSprite
+   * @param {Sprite} sprite
+   * @see p5.CollisionShape.prototype.getsDimensionsFromSprite
+   */
+  p5.CollisionCircle.prototype.updateFromSprite = function(sprite) {
+    if (this.getsDimensionsFromSprite) {
+      this.radius = Math.max(sprite.width, sprite.height)/2;
+    }
+    this.setParentTransform(sprite);
+  };
+
+  /**
+   * Recalculate cached properties, relevant vectors, etc. when at least one
+   * of the shape's transforms changes.  The base CollisionShape (and CollisionPoint)
+   * only need to recompute the shape's center, but other shapes may need to
+   * override this method and do additional recomputation.
+   * @method _onTransformChanged
+   * @protected
+   */
+  p5.CollisionCircle.prototype._onTransformChanged = function() {
+    p5.CollisionShape.prototype._onTransformChanged.call(this);
+    this._computeScaledRadius();
+  };
+
+  /**
+   * Call to update the cached scaled radius value.
+   * @method _computeScaledRadius
+   * @private
+   */
+  p5.CollisionCircle.prototype._computeScaledRadius = function() {
+    this._scaledRadius = new p5.Vector(this.radius, 0)
+      .transform(this._localTransform)
+      .transform(this._parentTransform)
+      .sub(this.center)
+      .mag();
+  };
+
+  /**
+   * Debug-draw this collision shape.
+   * @method draw
+   * @param {p5} sketch instance to use for drawing
+   */
+  p5.CollisionCircle.prototype.draw = function(sketch) {
+    sketch.push();
+    sketch.noFill();
+    sketch.stroke(0, 255, 0);
+    sketch.rectMode(sketch.CENTER);
+    sketch.ellipse(this.center.x, this.center.y, this._scaledRadius*2, this._scaledRadius*2);
+    sketch.pop();
+  };
+
+  /**
+   * Compute candidate separating axes relative to another object.
+   * @method _getCandidateAxes
+   * @protected
+   * @param {p5.CollisionShape} other
+   * @return {Array.<p5.Vector>}
+   */
+  p5.CollisionCircle.prototype._getCandidateAxes = function(other) {
+    // A circle has infinite potential candidate axes, so the ones we pick
+    // depend on what we're colliding against.
+
+    // TODO: Could generalize these two cases to other.vertices.map(v => sub(v, this.center))
+    if (other instanceof p5.OrientedBoundingBox || other instanceof p5.AxisAlignedBoundingBox) {
+      // There are four possible separating axes with a box - one for each
+      // of its vertices, through the center of the circle.
+      // TODO: Can we optimize and only check the nearest vertex?
+      var halfDiagonals = other.halfDiagonals;
+      return [
+        p5.Vector.sub(p5.Vector.add(other.center, halfDiagonals[0]), this.center),
+        p5.Vector.sub(p5.Vector.add(other.center, halfDiagonals[1]), this.center),
+        p5.Vector.sub(p5.Vector.sub(other.center, halfDiagonals[0]), this.center),
+        p5.Vector.sub(p5.Vector.sub(other.center, halfDiagonals[1]), this.center)
+      ];
```

Comment:
There are two kludges here; I'm not sure either is a big deal (yet) but I'm open to suggestions.

* `CollisionCircle` knows about `OrientedBoundingBox` and `AxisAlignedBoundingBox` which is useful, but in a perfect would wouldn't be necessary.
* We're just checking an axis for every vertex of the target box, when in theory we only need to check the axis with the closest vertex.  My brain went dead right around the time I needed to quickly figure out which one is closer though - other implementations infer this from previous checks against the box's normals, but that fights the beautiful abstracting of only caring about the set of axes here.